test/v200/utils/common/command_test_utils.go

```diff
@@ -0,0 +1,212 @@
+		command = testDevFile.createExecCommand()
+		testDevFile.SetExecCommandValues(command)
+	} else if commandType == schema.CompositeCommandType {
+		command = testDevFile.createCompositeCommand()
+		testDevFile.SetCompositeCommandValues(command)
+	} else if commandType == schema.ApplyCommandType {
+		command = testDevFile.createApplyCommand()
+		testDevFile.SetApplyCommandValues(command)
+	}
+	return *command
+}
+
+// createExecCommand creates and returns an empty exec command in a schema structure
+func (testDevFile *TestDevfile) createExecCommand() *schema.Command {
+
+	LogInfoMessage("Create an exec command :")
+	command := schema.Command{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+	command.Exec = &schema.ExecCommand{}
+	testDevFile.commandAdded(command)
+	return &command
+
+}
+
+// SetExecCommandValues randomly sets exec command attribute to random values
+func (testDevFile *TestDevfile) SetExecCommandValues(command *schema.Command) {
+
+	execCommand := command.Exec
+
+	// exec command must be mentioned by a container component
+	execCommand.Component = testDevFile.GetContainerName()
+
+	execCommand.CommandLine = GetRandomString(4, false) + " " + GetRandomString(4, false)
+	LogInfoMessage(fmt.Sprintf("....... commandLine: %s", execCommand.CommandLine))
+
+	// If group already leave it to make sure defaults are not deleted or added
+	if execCommand.Group == nil {
+		if GetRandomDecision(2, 1) {
+			execCommand.Group = testDevFile.addGroup()
+		}
+	}
+
+	if GetBinaryDecision() {
+		execCommand.Label = GetRandomString(12, false)
+		LogInfoMessage(fmt.Sprintf("....... label: %s", execCommand.Label))
+	} else {
+		execCommand.Label = ""
+	}
+
+	if GetBinaryDecision() {
+		execCommand.WorkingDir = "./tmp"
+		LogInfoMessage(fmt.Sprintf("....... WorkingDir: %s", execCommand.WorkingDir))
+	} else {
+		execCommand.WorkingDir = ""
+	}
+
+	execCommand.HotReloadCapable = GetBinaryDecision()
+	LogInfoMessage(fmt.Sprintf("....... HotReloadCapable: %t", execCommand.HotReloadCapable))
+
+	if GetBinaryDecision() {
+		execCommand.Env = addEnv(GetRandomNumber(1, 4))
+	} else {
+		execCommand.Env = nil
+	}
+	testDevFile.commandUpdated(*command)
+
+}
+
+// createCompositeCommand creates an empty composite command in a schema structure
+func (testDevFile *TestDevfile) createCompositeCommand() *schema.Command {
+
+	LogInfoMessage("Create a composite command :")
+	command := schema.Command{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+	command.Composite = &schema.CompositeCommand{}
+	testDevFile.commandAdded(command)
+
+	return &command
+}
+
+// SetCompositeCommandValues randomly sets composite command attribute to random values
+func (testDevFile *TestDevfile) SetCompositeCommandValues(command *schema.Command) {
+
+	compositeCommand := command.Composite
+	numCommands := GetRandomNumber(1, 3)
+
+	for i := 0; i < numCommands; i++ {
```

Comment:
same as: https://github.com/devfile/api/pull/348/files#r579475204
in golang, we use `range of` if index is not required in the scope

Reply:
Also in this case I am adding entries to a map, not iterating through it, so I don't think range works.

Reply:
thanks for the explanation, I did not check the loop clearer